build/cluster-operator-ansible/cluster-api-prep/deploy-cluster-api.yaml

```diff
@@ -0,0 +1,152 @@
+---
+- name: Setup the master node group
+  hosts: localhost
+  tasks:
+  - import_role:
+      name: openshift_aws
+      tasks_from: setup_master_group.yml
+
+#- name: set the master facts for hostname to elb
+#  hosts: masters
+#  gather_facts: no
+#  remote_user: root
+#  tasks:
+#  - import_role:
+#      name: openshift_aws
+#      tasks_from: master_facts.yml
+
+#- name: run the init
+#  import_playbook: ../init/main.yml
+
+- name: install cluster-api
+  hosts: masters[1]
+  gather_facts: no
+  remote_user: root
+  vars:
+    cluster_api_namespace: kube-cluster
+  tasks:
+  - name: import lib_openshift
+    import_role:
+      name: lib_openshift
+
+  - name: slurp local template content
+    slurp:
+      src: files/cluster-api-template.yaml
+    delegate_to: localhost
+    register: template
+
+  - name: "create {{ cluster_api_namespace }} namesapce"
+    oc_project:
+      state: present
+      name: "{{ cluster_api_namespace }}"
+      display_name: "Cluster API"
+      node_selector:
+      - node-role.kubernetes.io/master=true
+
+  - name: "create cluster-api controller manager service account"
+    oc_serviceaccount:
+      namespace: "{{ cluster_api_namespace }}"
+      state: present
+      name: cluster-api-controller-manager
+      #register: service_account_results
+
+  - name: "create cluster-api service account"
+    oc_serviceaccount:
+      namespace: "{{ cluster_api_namespace }}"
+      state: present
+      name: cluster-api-apiserver
+    register: service_account_results
+
+  - name: save secret with token
+    set_fact:
+      token_secret_name: "{{ item.name }}"
+    when: '"token" in item.name'
+    with_items: "{{ service_account_results.results.results[0].secrets }}"
+
+  - name: get token
+    oc_secret:
+      namespace: "{{ cluster_api_namespace }}"
+      state: list
+      name: "{{ token_secret_name }}"
+      decode: true
+    register: token
+
+  - name: create certificates
+    when: cluster_api_cert is not defined
+    block:
+    - name: create certificates
+      oc_adm_ca_server_cert:
+        signer_cert: /etc/origin/master/ca.crt
+        signer_key: /etc/origin/master/ca.key
+        signer_serial: /etc/origin/master/ca.serial.txt
+        hostnames: "clusterapi"
```

Comment:
For hostnames, use the service name + namespace as in https://github.com/openshift/cluster-operator/blob/master/contrib/apiserver-aggregation-tls-setup.sh#L24
`clusterapi.kube-cluster` and `clusterapi.kube-cluster.svc`